src/app/toolkit/add-tool/components/AddToolInputs.tsx

```diff
@@ -1,109 +1,148 @@
+            }
+          }
+        }
 
-    try {
-      console.log(`Inserting into database`);
-
-      DatabaseManager.addToDb("toolkit_items", {
-        id: crypto.randomUUID(),
-        name: formState.name,
-        categories: formState.categories,
-        description: formState.description,
-        checked: false,
-        infoUrl: formState.infoUrl,
-        imageUrl: formState.imageUrl,
-        timestamp: new Date().toISOString(),
-      });
-
-      console.log(`Created ${formState.name} in the database`);
-
-      // Show success message and wait for user acknowledgment
-      const userAcknowledged = window.confirm(
-        `Successfully added ${formState.name} to your toolkit!\n\nClick OK to return to the toolkit.`
-      );
-
-      if (userAcknowledged) {
-        router.push("/toolkit");
+        await DatabaseManager.addToDb("toolkit_items", {
+          id: crypto.randomUUID(),
+          name: formState.name,
+          categories: formState.categories,
+          description: formState.description,
+          checked: false,
+          infoUrl: formState.infoUrl,
+          imageUrl: formState.imageUrl,
+          timestamp: new Date().toISOString(),
+        });
+  
+        console.log(`Created ${formState.name} in the database`);
+  
+        setConfirmationModalOpen(true);
+      } catch (error) {
+        console.error("Error submitting form:", error);
+        alert(
+          `Failed to save tool: ${
+            error instanceof Error ? error.message : "Unknown error"
+          }`
+        );
       }
-    } catch (error) {
-      console.error("Error submitting form:", error);
-      alert(
-        `Failed to save tool: ${
-          error instanceof Error ? error.message : "Unknown error"
-        }`
-      );
-    }
-  };
+    };
+  
+    return (
+      <Button
+        label="Add Tool"
+        onClick={handleSubmit}
+        className="w-full mt-4 bg-twd-primary-purple "
+      />
+    );
+  }
 
   return (
-    <Button
-      label="Add Tool"
-      onClick={handleSubmit}
-      className="w-full mt-4 bg-twd-primary-purple "
-    />
-  );
-}
+    <div className="space-y-4 p-4">
+      <AddName />
+      <AddTags />
+      <AddDescription />
+      <AddImageUrl />
+      <AddInfoUrl />
+      <SubmitButton />
 
-export default function Inputs() {
-  return (
-    <AddToolProvider>
-      <div className="space-y-4 p-4">
-        <AddName />
-        <AddTags />
-        <AddDescription />
-        <AddImageUrl />
-        <AddInfoUrl />
-        <SubmitButton />
-      </div>
-    </AddToolProvider>
+      <Modal
+        title="Tool Added"
+        modalOpen={confirmationModalOpen}
+        forwardButton={confirmationForwardButton}
+      />
+
+      <Modal
+        title="You created a category but didn't use it. Would you like to save it anyway?"
+        modalOpen={unusedCategoryModalOpen}
+        forwardButton={unusedCategoryForwardButton}
+        backButton={unusedCategoryBackButton}
```

Comment:
The screenshot you posted doesn't fit the style of the rest of the app. There also isn't enough whitespace between the heading and the buttons. Please fix this (could be an idea to refactor the Modal component to use `text-lg` instead of `text-xl` for the buttons)

Reply:
Apologies, I thought Modal's inherent styling was app style and would be tweaked at the shared component level for consistency